envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/config/LocalReplyConfigFactory.kt

```diff
@@ -0,0 +1,262 @@
+    private fun setStatusCode(
+        matcherAndMapper: MatcherAndMapper,
+        responseMapperBuilder: ResponseMapper.Builder
+    ) {
+        if (matcherAndMapper.statusCodeToReturn != 0) {
+            responseMapperBuilder.setStatusCode(UInt32Value.newBuilder().setValue(matcherAndMapper.statusCodeToReturn))
+        }
+    }
+
+    private fun setResponseBody(
+        matcherAndMapper: MatcherAndMapper,
+        responseMapperBuilder: ResponseMapper.Builder
+    ) {
+        if (matcherAndMapper.bodyToReturn.isNotEmpty()) {
+            responseMapperBuilder.setBody(DataSource.newBuilder().setInlineString(matcherAndMapper.bodyToReturn))
+        }
+    }
+
+    private fun createStatusCodeFilter(statusCode: String): StatusCodeFilter.Builder {
+        val parsedStatusCodeMatcher = StatusCodeFilterParser.parseStatusCodeFilter(statusCode)
+        return StatusCodeFilter.newBuilder().setComparison(
+            ComparisonFilter.newBuilder()
+                .setOp(parsedStatusCodeMatcher.comparisonOperator)
+                .setValue(
+                    RuntimeUInt32.newBuilder()
+                        .setDefaultValue(parsedStatusCodeMatcher.comparisonCode)
+                        .setRuntimeKey("local_reply_mapper_http_code")
+                        .build()
+                )
+        )
+    }
+
+    private fun createResponseFlagFilter(responseFlags: List<String>): ResponseFlagFilter.Builder {
+        return ResponseFlagFilter.newBuilder().addAllFlags(responseFlags)
+    }
+
+    private fun createHeaderFilter(headerMatcher: HeaderMatcher): HeaderFilter.Builder {
+        val headerFilterBuilder = HeaderFilter.newBuilder()
+        val headerMatcherBuilder = io.envoyproxy.envoy.config.route.v3.HeaderMatcher.newBuilder()
+            .setName(headerMatcher.name)
+        return when {
+            headerMatcher.regexMatch.isNotEmpty() -> {
+                headerFilterBuilder.setHeader(
+                    headerMatcherBuilder.setSafeRegexMatch(
+                        RegexMatcher.newBuilder()
+                            .setGoogleRe2(RegexMatcher.GoogleRE2.getDefaultInstance())
+                            .setRegex(headerMatcher.regexMatch)
+                    )
+                )
+            }
+            headerMatcher.exactMatch.isNotEmpty() -> {
+                headerFilterBuilder.setHeader(headerMatcherBuilder.setExactMatch(headerMatcher.exactMatch))
+            }
+            else -> {
+                headerFilterBuilder.setHeader(headerMatcherBuilder.setPresentMatch(true))
+            }
+        }
+    }
+
+    @SuppressWarnings("ReturnCount")
+    private fun createResponseFormat(responseFormat: ResponseFormat): SubstitutionFormatString? {
+        val responseFormatBuilder = SubstitutionFormatString.newBuilder()
+        if (responseFormat.contentType.isNotEmpty()) {
+            responseFormatBuilder.contentType = responseFormat.contentType
+        }
+        if (responseFormat.textFormat.isNotEmpty()) {
```

Comment:
From this if downwards you can use `return if ...` to get rid of `@SuppressWarnings("ReturnCount")`. Maybe the "return count" rule is stupid and we should get rid of it?